src/App.tsx

```diff
@@ -1,9 +1,30 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
 export const App: React.FC = () => {
+  const [query, setQuery] = useState('');
+  const [visibleMovies, setVisibleMovies] = useState(moviesFromServer);
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+  };
+
+  useEffect(() => {
+    const filterByElement = (elem: string) => {
```

Comment:
move it from useEffect
It's bad way to update your list
Also, move this `filterByElement` from the component
Also, split this `elem.toLowerCase().includes(query.toLowerCase().trim());` with variables